content/news/2023-06-from-single-files-to-collections/index.md

```diff
@@ -0,0 +1,150 @@
+
+## Intermediate workflow specific solution
+
+### Filter the collection
+
+Collections can be filtered for failed or empty datasets using collection tools such as [filter empty datasets](https://usegalaxy.eu/?tool_id=__FILTER_EMPTY_DATASETS__&version=1.0.0) and [filter failed datasets](https://usegalaxy.eu/?tool_id=__FILTER_FAILED_DATASETS__&version=1.0.0). 
+
+<div class="center">
+<div class="img-sizer" style="width: 100%">
+
+![Empty collection](./figs/empty_collection.png)
+
+</div>  
+<figcaption>
+  Filter failed collection elements
+</figcaption>
+</div>
+
+<div class="center">
+<div class="img-sizer" style="width: 100%">
+
+![Empty collection](./figs/failed_collection.png)
+
+</div>  
+<figcaption>
+  Filter empty collection elements
+</figcaption>
+</div>
+
+
+Although this can solve the issue in some cases immediately, further considerations need to be made. 
+First, often one cannot really know at which step the collection will be affected.
+To cover all cases one would need to add the filter steps for every produced collection, which will increase the workflow steps unreasonably. 
+Secondly, the filtering will change the size of the collection. If downstream tools depend on a specific collection size, which is always the case if a tool
+takes two or more collections as input, the tool will also fail. That's basically a follow up problem of the first problem.
+This case can still be solved by a intermediate step where the second collection is basically reindexed by the same element identifiers then the collection with
+missing elements. 
+
+<div class="center">
+<div class="img-sizer" style="width: 100%">
+
+![Empty collection](./figs/reindex_collection.png)
+
+</div>  
+<figcaption>
+  Proposed solution to reindexed second collection using elements of the first collection.
+</figcaption>
+</div>  
+
+However, if this logic needs to be applied to multiple collections or again for multiple steps, the workflow becomes even more unreasonably large and complex. 
+
+Furthermore it needs to be considered, that filtering empty or failed elements from a collection
+can hide the root cause of the problem. The question why some tool runs fail or produce empty output should always be further investigated.
+
+### Rerun the workflow with rerun activated 
+
+In some cases a rerun of the workflow can also solve the issue, for example in our use case where the failure was rather randomly.
+For this strategy it is beneficial to activate the rerun option of galaxy (User/Preferences/Manage Information) on instance with version > 23.0. 
+This allows to only rerun failed elements, which saves computing resources and time.
```

Comment:
That's also true if you click on rerun and replace collection element, I think I would say

This creates a new workflow run where successful outputs are copied from the previous execution instead of being run again, and only failed jobs and jobs depending on the failed outputs will be submitted again.